rust/src/main.rs

```diff
@@ -145,7 +129,53 @@ fn gen(args: Args) -> Result<()> {
     documents.seek(SeekFrom::Start(0))?;
     encode_sequence(&mut documents, 1, [num_documents].iter().copied())?;
 
+    eprintln!("Processing document lengths...");
+
+    let mut sizes = BufWriter::new(File::create(format!("{}.sizes", args.output))?);
+    let mut trecids = BufWriter::new(File::create(format!("{}.documents", args.output))?);
+
+    let bar = ProgressBar::new(num_documents as u64);
+    bar.set_style(pb_style());
+    bar.set_draw_delta(num_documents as u64 / 100);
+    sizes.write_all(&num_documents.to_ne_bytes())?;
+
+    let mut docs_seen: usize = 0;
+    let expected_docs: usize = header
+                          .get_num_docs()
+                          .to_usize()
+                          .ok_or_else(|| format!("Cannot cast to usize: {}", header.get_num_docs()))?;
+
+
+    while docs_seen < expected_docs {
```

Comment:
Again, write rather:
```rust
for docs_seen in 0..expected_docs {
```
instead of introducing a variable earlier and then incrementing it manually.